localstack/services/s3/cors.py

```diff
@@ -31,10 +32,10 @@
 
 class S3CorsHandler(Handler):
 
-    bucket_cors_index: BucketCorsIndex
+    bucket_cors_index: BucketCorsIndex | BucketCorsIndexV2
 
-    def __init__(self):
-        self.bucket_cors_index = BucketCorsIndex()
+    def __init__(self, bucket_cors_index: BucketCorsIndex | BucketCorsIndexV2 = None):
+        self.bucket_cors_index = bucket_cors_index or BucketCorsIndex()
```

Comment:
This is a minor thing, but since `S3CorsHandler` seems independent from the provider, maybe we could further remove the dependency here, instead of introducing a dependency to _both_ providers.
Suggestion (note: remove the call to `BucketCorsIndex()`):

introduce a Protocol here:
```python
class BucketCorsIndex(Protocol):

    @property
    def cors(self) -> dict[str, CORSConfiguration]:
        raise NotImplementedError

    @property
    def buckets(self) -> set[str]:
        raise NotImplementedError


class S3CorsHandler(Handler):
    bucket_cors_index: BucketCorsIndex

    def __init__(self, bucket_cors_index: BucketCorsIndex):
        self.bucket_cors_index = bucket_cors_index
        ...
```

then, make sure to instantiate the provider-specific BucketCorsIndex in the provier: https://github.com/localstack/localstack/blob/a7e97ea120cd37e90ce0722b241d1b7d82f34582/localstack/services/s3/provider.py#L249